src/stp/plays/PenaltyThemPrepare.cpp

```diff
@@ -28,23 +28,23 @@ PenaltyThemPrepare::PenaltyThemPrepare() : Play() {
 uint8_t PenaltyThemPrepare::score(world::World* world) noexcept { return 100; }
 
 void PenaltyThemPrepare::calculateInfoForRoles() noexcept {
-    auto width = field.getFieldWidth();
-    auto length = field.getFieldLength();
+    const auto xPosition = -4 * control_constants::ROBOT_RADIUS;
+    const auto yPosition = Constants::STD_TIMEOUT_TO_TOP() ? field.getFieldWidth() / 2.2 : -field.getFieldWidth() / 2.2;
 
     // Keeper
     stpInfos["keeper"].setPositionToMoveTo(Vector2(field.getOurGoalCenter()));
 
     // regular bots
-    stpInfos["formation_0"].setPositionToMoveTo(Vector2(-length / 4 + 2, width / 8));
-    stpInfos["formation_1"].setPositionToMoveTo(Vector2(-length / 4 + 2, -width / 8));
-    stpInfos["formation_2"].setPositionToMoveTo(Vector2(-length / 8 + 2, width / 4));
-    stpInfos["formation_3"].setPositionToMoveTo(Vector2(-length / 8 + 2, -width / 4));
-    stpInfos["formation_4"].setPositionToMoveTo(Vector2(-length * 3 / 8 + 2, 0.0));
-    stpInfos["formation_5"].setPositionToMoveTo(Vector2(-length * 3 / 8 + 2, width / 5));
-    stpInfos["formation_6"].setPositionToMoveTo(Vector2(-length * 3 / 8 + 2, -width / 5));
-    stpInfos["formation_7"].setPositionToMoveTo(Vector2(-length / 4 + 2, width / 3));
-    stpInfos["formation_8"].setPositionToMoveTo(Vector2(-length / 4 + 2, -width / 3));
-    stpInfos["formation_9"].setPositionToMoveTo(Vector2(-length / 8 + 2, 0.0));
+    stpInfos["formation_0"].setPositionToMoveTo(Vector2(xPosition, yPosition));
+    stpInfos["formation_1"].setPositionToMoveTo(Vector2(xPosition - 4 * control_constants::ROBOT_RADIUS, yPosition));
```

Comment:
Oh just saw this, maybe xPosition, then 2*xPosition, 3*xPosition, etc is nicer? Because what you're subtracting every time is 1 more xPosition

Reply:
Or start with xPosition = 0, and then call the '-4*radius' thing an offset and subtract an extra offset every time? Idk, just a style thing not really important

Reply:
I copied this part of the code from the time out play, and I think you make a good point but if we use 2xPosition etc, it's maybe better to use a for-loop to remove double code, so ive implemented that now (also in time out)

Reply:
Even nicer, that looks satisfying :+1: